src/components/DecisioningEngine/createIndexedDB.js

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { HISTORICAL_DATA_STORE } from "./constants";
+
+const dbName = HISTORICAL_DATA_STORE;
+
+export default () => {
+  let db;
+  const DB_INITIALIZATION_TIMEOUT = 200;
+
+  const setupIndexedDB = () => {
+    return new Promise((resolve, reject) => {
```

Comment:
[In the Identity component,](https://github.com/adobe/alloy/blob/main/src/components/Identity/injectAwaitIdentityCookie.js) we have a similar situation: we need to wait for the identity cookie before we firing off other requests. However, the `awaitIdentityCookie()` is implemented with Promises. 

I think if you use Promises here, it would make the implementation a little neater and removes the need for `waitForDBInitialization()` and it's `setTimeout()`.

For example, if you write `setupIndexDB()` like so…

```js
export default () => {
  let dbPromise;
  const setupIndexedDB = () => {
    dbPromise = new Promise((resolve, reject) => {
      // set up login
    });
    // no need to expose the DB outside of this module
    return dbPromise.then(() => {});
  };
}
```

Then you can use `dbPromise` like this…

```js
const addRecord = record => {
  return dbPromise.then(db => {
    const transaction = db.transaction("events", "readwrite");
    const objectStore = transaction.objectStore("events");
    const objectStoreRequest = objectStore.add(record);
    return new Promise((resolve, reject) => {
      objectStoreRequest.onsuccess = () => resolve(true);
      objectStoreRequest.onerror = (txEvent) => reject(txEvent.target.error);
    });
  })
};
```

This way, you don't need `waitForDBInitialization()` with it's `setTimeout()` or `getIndexDB()` (which are kind of a code smells, in my opinion).